lib/ui/survey_details/survey_details_screen.dart

```diff
@@ -0,0 +1,151 @@
+import 'package:survey_flutter_ic/ui/survey_details/survey_details_ui_model.dart';
+import 'package:survey_flutter_ic/ui/survey_details/survey_details_view_model.dart';
+import 'package:survey_flutter_ic/ui/survey_details/survey_details_state.dart';
+import 'package:survey_flutter_ic/usecases/get_survey_details_use_case.dart';
+import 'package:survey_flutter_ic/utils/context_ext.dart';
+import 'package:survey_flutter_ic/utils/custom_app_bar.dart';
+import 'package:survey_flutter_ic/utils/dimension.dart';
+
+final surveyDetailsViewModelProvider = StateNotifierProvider.autoDispose<
+    SurveyDetailsViewModel, SurveyDetailsState>(
+  (_) => SurveyDetailsViewModel(getIt.get<GetSurveyDetailsUseCase>()),
+);
+
+class SurveyDetailsScreen extends ConsumerStatefulWidget {
+  final String surveyId;
+
+  const SurveyDetailsScreen({
+    required this.surveyId,
+    super.key,
+  });
+
+  @override
+  ConsumerState<SurveyDetailsScreen> createState() =>
+      SurveyDetailsScreenState();
+}
+
+class SurveyDetailsScreenState extends ConsumerState<SurveyDetailsScreen> {
+  AppBar get _appBar => CustomAppBar.backButton(context: context);
+
+  Widget get _background => SizedBox(
+        width: MediaQuery.of(context).size.width,
+        height: MediaQuery.of(context).size.height,
+        child: Consumer(
+          builder: (_, ref, __) {
+            final state = ref.watch(surveyDetailsViewModelProvider);
+            return state.maybeWhen(
+              orElse: () => _defaultBackground,
+              success: (uiModel) => FadeInImage.assetNetwork(
+                placeholder: Assets.images.bgLoginOverlay.path,
+                image: uiModel.imageUrl,
+                fit: BoxFit.cover,
+                alignment: Alignment.center,
+              ),
+            );
+          },
+        ),
+      );
+
+  Widget get _defaultBackground => Image(
+        image: Assets.images.bgLoginOverlay.image().image,
+        fit: BoxFit.cover,
+      );
+
+  Widget get _emptyBody => Container();
+
+  Widget get _body => Consumer(
+        builder: (_, ref, __) {
+          final state = ref.watch(surveyDetailsViewModelProvider);
+          return state.maybeWhen(
+            orElse: () => _emptyBody,
+            loading: () => const SurveyDetailsSkeletonLoading(),
+            success: (uiModel) => _surveyDetails(uiModel),
+          );
+        },
+      );
+
+  Widget _surveyDetails(SurveyDetailsUIModel uiModel) => Consumer(
+        builder: (_, ref, __) {
+          return Container(
+            width: MediaQuery.of(context).size.width,
+            padding: const EdgeInsets.all(Dimensions.paddingMedium),
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.start,
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                Text(
+                  uiModel.title,
+                  style: Theme.of(context).textTheme.displayLarge,
```

Comment:
Please define `textTheme` as a part of `BuildContextExtension`.

```suggestion
                  style: context.textTheme.displayLarge,
```